NFTBrowser/ImageDownloader.swift

```diff
@@ -0,0 +1,123 @@
+import Foundation
+import UIKit
+import os
+
+class ImageDownloader {
+            
+    let inMemoryCache: InMemoryCache = {
+        return InMemoryCache()
+    }()
+    
+    let session: URLSession = {
+        let config = URLSessionConfiguration.default
+        config.requestCachePolicy = .returnCacheDataElseLoad
+        return URLSession(configuration: config)
```

Comment:
Add URLCache with diskCache and, possibly, memoryCache. Set the size of the diskCache to 100x of screen size or so